subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -808,4 +790,137 @@ static uint8_t set_cte_info_in_per_adv_chain(struct ll_adv_set *adv, struct lll_
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, 0, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, NULL, &pdu_prev, &pdu,
+				     &extra_data_prev, &extra_data, ter_idx);
+	if (err) {
+		return err;
+	}
+
+	if (extra_data) {
+		ull_adv_sync_extra_data_set_clear(extra_data_prev,
+						  extra_data,
+						  ULL_ADV_PDU_HDR_FIELD_CTE_INFO,
+						  0, NULL);
+	}
+
+	pdu_rem_field_flags = ULL_ADV_PDU_HDR_FIELD_CTE_INFO;
+
+#if defined(CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX) && (CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1)
+	struct pdu_adv *pdu_new, *pdu_chained;
+	bool new_chain;
+
+	/* It is possible that the function is called after e.g. advertising data were updated.
+	 * In such situation the function will run on already allocated chain. Do not allocate
+	 * new chain then. Reuse already allocated PDUs and allocate new ones only if the chain
+	 * was not updated yet.
+	 */
+	new_chain = (pdu_prev == pdu ? false : true);
+
+	/* Get next PDU in a chain. Alway use pdu_prev because it points to actual
+	 * former chain.
+	 */
+	pdu_chained = lll_adv_pdu_linked_next_get(pdu_prev);
+
+	/* Go through existing chain and remove CTE info. */
+	while (pdu_chained != NULL) {
+		if (pdu_ext_adv_is_empty_without_cte(pdu_chained) == true) {
+			/* If there is an empty PDU then all remaining PDUs shoudl be released. */
+			lll_adv_pdu_linked_release_all(pdu_chained);
+			pdu_chained = NULL;
+			/* Set new end of chain in PDUs linked list. */
+			lll_adv_pdu_linked_append(NULL, pdu);
+		} else {
+			/* Update one before pdu_chained */
+			err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu, 0,
+							 pdu_rem_field_flags, NULL);
+			if (err != 0) {
+				/* TODO: return here leaves periodic advertising chain in
+				 * an inconsisten state. Add gracefull return or assert.
+				 */
+				return err;
+			}
+
+			/* Prepare for next iteration. Allocate new PDU or move to next one in
+			 * a chain.
+			 */
+			if (new_chain) {
+				pdu_new = lll_adv_pdu_alloc_pdu_adv();
+				lll_adv_pdu_linked_append(pdu_new, pdu);
+				pdu = pdu_new;
+			} else {
+				pdu = lll_adv_pdu_linked_next_get(pdu);
+			}
+
+			/* Move to next chained PDU (it moves through chain that is in use
+			 * by LLL or is new one with updated advertising payload).
+			 */
+			pdu_prev = pdu_chained;
+			pdu_chained = lll_adv_pdu_linked_next_get(pdu_prev);
+		}
+	}
+
+	/* Update last PDU in a chain. It may not have aux_ptr.
+	 * NOTE: If there is no AuxPtr flag in the PDU, attempt to remove it does not harm.
+	 */
+	pdu_rem_field_flags |= ULL_ADV_PDU_HDR_FIELD_AUX_PTR;
+#endif /* CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX && CONFIG_BT_CTLR_DF_PER_ADV_CTE_NUM_MAX > 1 */
+
+	err = ull_adv_sync_pdu_set_clear(lll_sync, pdu_prev, pdu, 0, pdu_rem_field_flags, NULL);
+	if (err != 0) {
+		/* TODO: return here leaves periodic advertising chain in an inconsisten state.
+		 * Add gracefull return or assert.
+		 */
+		return err;
+	}
+	return 0;
```

Comment:
It often increases readability (slightly) if there is an empty line before `return` statements after a block like this

Reply:
Added empty line before `return`